src/database/sql_database.cc

```diff
@@ -914,17 +914,25 @@ std::vector<std::shared_ptr<CdsObject>> SQLDatabase::browse(BrowseParam& param)
     commit("browse");
 
     std::vector<std::shared_ptr<CdsObject>> result;
+    std::vector<int> contIds;
```

Comment:
If you use a map<int, CdsContainer> you can avoid the second check and run through it directly.

Reply:
Before, I had the collection of contIds in a separate loop but merged them together. As an alternative, we could make a second CdsObject-vector of containers and use this to iterate in the second loop. I will definitely put some more work into this PR but currently my spare time is limited. There's still time for 1.9.3 :)

Reply:
I added a new changeset which addesses this. We could move the extraction of ID values into `SQLDatabase::getChildCounts` but on the other hand, I prefer to keep `SQLDatabase::getChildCount` and `SQLDatabase::getChildCounts` very similar.